### PR TITLE
fix lint tests

### PR DESCRIPTION
**Pull request by dhood**

### CODE REVIEW

**Comment by dhood** [comment]
dealt with in https://github.com/ros2/rclpy/pull/13

**Comment by dirk-thomas** [comment]
Since @wjwwood removed the patch from his PR (https://github.com/ros2/rclpy/pull/12#issuecomment-206088575) this can be reopened.

File: rclpy/rclpy/impl/rmw_implementation_tools.py [hunk 1]
```diff
@@ -30,8 +30,10 @@ def reload_rmw_implementations():
     global __rmw_implementations
     __rmw_implementations = sorted(ament_index_python.get_resources('rmw_implementation').keys())
 
-    # Remove implementations that are being filtered for in the rclp CMakeLists so they cannot be selected
-    __rmw_implementations = [rmw_impl for rmw_impl in __rmw_implementations if rmw_impl not in ['rmw_connext_dynamic_cpp', 'rmw_fastrtps_cpp']]
+    # Remove implementations that are being filtered for in the rclpy CMakeLists so 
+    # that they cannot be selected
+    __rmw_implementations = [rmw_impl for rmw_impl in __rmw_implementations 
+            if rmw_impl not in ['rmw_connext_dynamic_cpp', 'rmw_fastrtps_cpp']]
```
Comment (dirk-thomas) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: For PEP8 compliance can you modify the wrapping and indentation to:

```
__rmw_implementations = [
    rmw_impl for rmw_impl in __rmw_implementations 
    if rmw_impl not in ['rmw_connext_dynamic_cpp', 'rmw_fastrtps_cpp']]
```

**Comment by wjwwood** [comment]
We can certainly merge this before mine. +1 (with @dirk-thomas's comments addressed).

**Comment by dhood** [comment]
ok fixed. Sorry about this pep8 stuff - something is missing on my system so that the lint checks are just spitting out junk (around 1000 lines just for `rclpy`). I'll find whatever is missing today and add it to the ros2 install instructions.

File: rclpy/rclpy/impl/rmw_implementation_tools.py [hunk 2]
```diff
@@ -85,5 +88,16 @@ def import_rmw_implementation():
...
+        __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
+    except ImportError:
+        logger = logging.getLogger('rclpy')
+        logger.error(
+            "The rmw implementation selected '{0}' is not one of the available implementations: "
+            "{1}. Please specify another with 'RCLPY_IMPLEMENTATION'."
+            .format(__selected_rmw_implementation, get_rmw_implementations())
+        )
+        raise InvalidRCLPYImplementation()
+    '''
```
Comment (dirk-thomas) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: What about this block?

File: rclpy/rclpy/impl/rmw_implementation_tools.py [hunk 3]
```diff
@@ -85,5 +88,4 @@ def import_rmw_implementation():
     module_name = '._rclpy__{rmw_implementation}'.format(
         rmw_implementation=__selected_rmw_implementation,
     )
-    __rmw_implementation_module = importlib.import_module(module_name, package='rclpy')
```
Comment (wjwwood) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: I think you dropped this one unintentionally? :stuck_out_tongue:

Comment (dhood) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: ah, you guys are too fast (and I have not been paying close enough attention - sorry for wasting your time on this.)

File: rclpy/rclpy/impl/rmw_implementation_tools.py [hunk 4]
```diff
@@ -30,8 +30,11 @@ def reload_rmw_implementations():
     global __rmw_implementations
     __rmw_implementations = sorted(ament_index_python.get_resources('rmw_implementation').keys())
 
-    # Remove implementations that are being filtered for in the rclp CMakeLists so they cannot be selected
-    __rmw_implementations = [rmw_impl for rmw_impl in __rmw_implementations if rmw_impl not in ['rmw_connext_dynamic_cpp', 'rmw_fastrtps_cpp']]
+    # Remove implementations that are being filtered for in the rclpy CMakeLists so
```
Comment (dirk-thomas) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: Nitpick: I would wrap after `CMakeLists` then the next line makes more sense on its own.

**Comment by dirk-thomas** [comment]
+1

File: rclpy/rclpy/impl/rmw_implementation_tools.py (hunk 4 above)
Comment (dhood) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: thank you, I'll do that from now on

File: rclpy/rclpy/impl/rmw_implementation_tools.py (hunk 3 above)
Comment (wjwwood) [inline rclpy/rclpy/impl/rmw_implementation_tools.py]: No worries, we're supposed to wait until it is in review (i.e. it has the `in review` tag) anyways :smile:.

**Comment by wjwwood** [comment]
lgtm too

**Comment by dhood** [comment]
Regarding my previous comment re linter tests failing, I found (in addition to https://github.com/ament/ament_lint/issues/48) that in the CI build there's a line
```
Step 14 : RUN apt-get update && apt-get install -y clang-format-3.4 cppcheck git pyflakes python3-coverage python3-mock python3-nose python3-pep8 uncrustify
```
which is (mostly) missing from the [source installation instructions](https://github.com/ros2/ros2/wiki/Linux-Development-Setup), so I was missing some of the packages.

Since the instructions call ament with `--build-tests`, should we also include a line in the instructions for the appropriate packages to *run* the tests? Or is there a more appropriate way to fix this?

**Comment by wjwwood** [comment]
@dhood I think you're right, it appears to be missing from both the Linux and OS X development guides. Ironically I think the Windows setup guide is up-to-date: https://github.com/ros2/ros2/wiki/Windows-Development-Setup#installing-a-few-dependencies

It's probably because our Linux and OS X machines are already set up with this tools for our other work, whereas the Windows machines are otherwise unused. Can you update those wiki pages (Linux and OS X)? If you need help or want a review on your changes, just let me know.

**Pull request merged**